src/escaped.rs

```diff
@@ -0,0 +1,25 @@
+#[inline(always)]
+pub(crate) fn escape_char(c: char) -> Option<&'static str> {
```

Comment:
If this returned a String, every time one of these characters was returned, it would need to allocate a new string, then copy in the &'static str... so in order to return it as a string and use it in a string, it would be &'static str -> String (allocation) -> Add to string (possible allocation). This way it's &'static str -> Add to string (possible allocation).

Reply:
In terms of Option vs Result, that's a good point, but I'm not sure it's worth the extra allocation.